Basic Transformation Implementation

todouu opened the pull request:
This branch develop the basic transformation function for the XZES40 program.
After complete this develop, program should do following things.

1. Program can successfully transformer XML file and XSL files to a new XML files.
2. Program can handle the parsed XML file and XSL files for future cache using.

fixes #5 
fixes #22

pop commented on xzes40/transformer/include/document.hpp:
```diff
@@ -19,12 +19,12 @@ namespace xzes
         public:
-            Document( );
-            int set_uri( uri_t );
+            Document( xzes::uri_t file_path );                //constructor
```
Please put this comment above this line and align them.

```c++
// constructor
Document( xzes::uri_t file_path );
```

pop commented on xzes40/transformer/include/document.hpp:
```diff
@@ -19,12 +19,12 @@ namespace xzes
-            Document( );
-            int set_uri( uri_t );
+            Document( xzes::uri_t file_path );                //constructor
+            void set_uri( uri_t );
```
Why are these type `void`? If they are type `int` we can return error codes.

pop commented on xzes40/transformer/src/document.cpp:
```diff
@@ -56,21 +67,20 @@ int xzes::Document::set_uri( xzes::uri_t file_path )
 // ----------------------------------------------------------------------------
 // int set_id( );
-//
+// change to void function
```
Why?

pop reviewed (changes requested):
# Feedback so far:

- Why convert int functions to void? This makes it harder to return status codes like SUCCESS or FAILURE.
- Consider adding a test to this code to make sure it works. A simple test that transforms two files would be good enough. We can discuss how to write tests at our meeting or in an issue.
- Please rename this Pull Request to something more useful like `Basic Transformation Implementation`.
- Add a `fixes #5` to your pull request description. This auto-closes the issue once this is merged.

pop committed:
Fixes the majority of bugs in this branch.

Also includes updates to license and scrapping most of the
transformation code in favor of a simpler approach that actually
compiles and runs.

pop commented:
@Todopen A few notes:

1. I am glad that you added the license to the files, but I really wanted @zixunlu to do this as he has not contributed anything to the code yet and needs something to do to make sure he understands how to contribute code with git.
2. Please try not to push code that fails to build. I am glad that you did work on this, but it is bad practice to push code that fails to build. Do you have your development environment setup? Are you able to test changes easily? I can help you set this up if you do not have it setup yet.
3. In vagrant set up git so that it commits as *your* user and not the default vagrant user. We can change that commit so the commit made by `Vagrant Default User` so that it is under your account. I will show you how soon.

I will fix this Pull Request to make sure we have an alpha release on time.

pop commented:
@Todopen I made a lot of changes to this PR. It still breaks ([see here](https://travis-ci.org/XZES40/XZES40-Transformer/builds/196474989#L225)).

@zixunlu I want you to figure out what is causing the break and fix it.

zixunlu commented:
Ok I will try to fix it tonight.

zixunlu commented:
I saw the errors which it looks similar like last time.

todouu committed:
Finalization of transformation code.

pop commented on xzes40/transformer/Makefile:
```diff
@@ -8,8 +8,13 @@
 # Thanks: http://www.cs.colby.edu/maxwell/courses/tutorials/maketutor/
 ###############################################################################
 
+<<<<<<< HEAD
```
Remove this from your commit. This is git garbage.

pop commented on xzes40/transformer/include/document.hpp:
```diff
@@ -8,7 +8,7 @@
 // http://www.apache.org/licenses/LICENSE-2.0
 //
-// Unless required by applicable law or agreed to in writing, software
+// Unless required by applicable law or agr6eed to in writing, software
```
remove 6.

pop commented on xzes40/transformer/src/transform.cpp:
```diff
@@ -41,13 +41,83 @@ XALAN_USING_XALAN(XSLTResultTarget);
 
 int xzes::transform_documents( xzes::cli_arguments_t *args )
 {
+<<<<<<< HEAD
```
remove this git garbage.

pop reviewed (changes requested):
You included a lot of git garbage. In the future please use `git fetch --all` and then `git rebase origin/branch-name` and follow the instructions that git give you.

For now remove the git garbage by choosing the version of the file you want (between `>>>` and `===` and `<<<`), then `git commit` and `git push origin branch-name`.

todouu committed:
Initial transformation functionality code
